packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue

```diff
@@ -466,61 +469,45 @@ export default defineComponent({
     }
   },
   watch: {
-    newExpiration(expiration) {
-      this.updateLink({
-        link: {
-          ...this.link,
-          expiration
-        }
-      })
+    newExpiration(expirationDateTime: string) {
+      this.$emit('updateLink', { linkShare: { ...this.linkShare, expirationDateTime } })
     }
   },
   methods: {
     deleteLink() {
-      this.$emit('removePublicLink', { link: this.link })
+      this.$emit('removePublicLink', { link: this.linkShare })
       ;(this.$refs.editPublicLinkDropdown as InstanceType<typeof OcDrop>).hide()
     },
     showRenameModal() {
       this.dispatchModal({
         title: this.$gettext('Edit name'),
         confirmText: this.$gettext('Save'),
         hasInput: true,
-        inputValue: this.link.name,
+        inputValue: this.linkShare.displayName,
         inputLabel: this.$gettext('Link name'),
         onInput: (name, setError) => {
           if (name.length > 255) {
             return setError(this.$gettext('Link name cannot exceed 255 characters'))
           }
           return setError(null)
         },
-        onConfirm: (name: string) =>
-          Promise.resolve(
-            this.updateLink({
-              link: {
-                ...this.link,
-                name
-              }
-            })
-          )
+        onConfirm: (displayName: string) => {
+          const linkShare = this.linkShare
+          linkShare.displayName = displayName
+          return Promise.resolve(this.$emit('updateLink', { linkShare }))
+        }
```

Comment:
This is completely sync, right?

Semantically it's also really weird to me to wrap the result of $emit in a promise - isn't it empty anyhow?